code/datums/mapgen/CaveGenerator.dm

```diff
@@ -137,6 +137,12 @@
 					can_spawn = FALSE
 					break
 
+			//we may have generated a plant on the target turf, so let's not place
+			//a geyser here if that's the case
+			for(var/obj/structure/turf_existing_feature in turf.contents)
```

Comment:
Can we just use locate() ? A for loop where we don't use it's var looks wasted. Maybe do

`can_spawn = isnull(locate(/obj/structure) in turf)`

Also the var `turf` is highlighted like a type In the editor since it is a type. Renaming it to something like `target_turff` would be appreciated.